src/hooks/useContextMenu.ts

```diff
@@ -73,18 +79,24 @@ export default function useContextMenu() {
 
   const handleFavoriteToggleMenuItemClick = async (
     selectedFiles: FileOrFolder[]
-  ): Promise<void> => {
-    if (currentFileSharePath) {
-      await handleFavoriteChange(
-        {
-          type: 'folder',
-          folderPath: selectedFiles[0].path,
-          fsp: currentFileSharePath
-        },
-        'folder'
-      );
+  ): Promise<Result<null>> => {
+    try {
+      if (currentFileSharePath) {
+        await handleFavoriteChange(
+          {
+            type: 'folder',
+            folderPath: selectedFiles[0].path,
+            fsp: currentFileSharePath
+          },
+          'folder'
+        );
+        setShowContextMenu(false);
+      }
+    } catch (error) {
+      logger.error(error);
+      return createFailureResult(parseError(error));
```

Comment:
If the error handling pattern will always compose these functions, then should we have a function for it? Something like `return handleError(error)` and then that function would log, get the message, and return the failure object.